### PR TITLE
4602issue heap

### PR DESCRIPTION
**Pull request by reerajput930**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

contributed by creating code for the formation of max-heap in the C++ section.

### CODE REVIEW

**Comment by reerajput930** [comment]
sir @HarshCasper, @siddharth25pandey, @bhav09, @ArchanGhosh,@ankitaggarwal23 
I have made the PR,
my file can be tested

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@reerajput930 Do not edit `.deepsource.toml` file. Please update the readme file. Also, provide the issue number.

File: .deepsource.toml [hunk 1]
```diff
@@ -6,19 +6,14 @@ enabled = true
...
   runtime_version = "3.x.x"
-  type_checker = "mypy"
 
 [[analyzers]]
 name = "go"
 enabled = true
 
   [analyzers.meta]
-  import_paths = ["github.com/TesseractCoding/NeoAlgo"]
+  import_paths = ["github.com/reerajput930/NeoAlgo"]
```
Comment (HarshCasper) [inline .deepsource.toml:15]: Why did you change this? Please don't do it.

File: C-Plus-Plus/heap/max_heap.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+#include <iostream>
+using namespace std;
+
+//function for comparison and swap process
+void swap_func(int *arr, int n, int i)
+{
+
+    int largest = i;       //parent node
+    int l = ((2 * i) + 1); //formula to access left child of parent node
+    int r = ((2 * i) + 2); //formula to access right child of parent node
```
Comment (HarshCasper) [inline C-Plus-Plus/heap/max_heap.cpp:14]: Don't use comments beside the line of code, but above it.

File: C-Plus-Plus/heap/max_heap.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    //parent comaparing with left child,
+    //first check if left child exit
```
Comment (HarshCasper) [inline C-Plus-Plus/heap/max_heap.cpp:17]: Use Multi-Line Comments

File: C-Plus-Plus/heap/max_heap.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    if (l <= n and arr[l] > arr[largest])
+        largest = l;
+    //parent comaparing with right child,
+    //first check if right child exit
```
Comment (HarshCasper) [inline C-Plus-Plus/heap/max_heap.cpp:21]: Use Multi-Line Comments

File: C-Plus-Plus/heap/max_heap.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+//function to prient the array which is max-heap
+void prient(int arr[], int n)
+{
+    for (int i = 0; i < n; i++)
+        cout << arr[i] << " ";
+}
+//driver code
+int main()
+{
+    int array[] = {4, 56, 2, 3, 7, 8};        // array
```
Comment (HarshCasper) [inline C-Plus-Plus/heap/max_heap.cpp:53]: Don't use static input. Use dynamic input instead

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Code plagiarized from [here](https://www.geeksforgeeks.org/heap-sort/).

**Comment by reerajput930** [comment]
@ankitaggarwal23 don't close it, The solution of this problem is unique, we can't make a different solution to this problem.
plz try to understand.

**Pull request closed without merging**